### ISSUE TITLE
Anilist Tracking error http 400

### ISSUE THREAD

**Issue by ExoZNiK** [tachiyomiorg/tachiyomi#9446]
### Steps to reproduce

1- click on a new manga that you added to you library
2- click tracking logo
3- choose anilist to add for tracking
4- http error 400 (with sad face included)

### Expected behavior

Instead of error 400 should come the list of manga that match the one you want to track. I didn't add a manga for a while but the feature befare worked nice. I can filter manga with anilist syncronized only and all my old manga still works sync. Maybe last update broke something

### Actual behavior

Http error 400

### Tachiyomi version

0.14.6

### Android version

Android 13

### Device

Realme GT Master Edition

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have gone through the [FAQ](https://tachiyomi.org/help/faq/) and [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.14.6](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by curche** [comment]
Try:
1. Log Out from anilist 
2. Relogin to anilist
3. try again

**Comment by ExoZNiK** [comment]
Ok seems dumb but it worked. I didn't remember the password at first so i didn't tried in first place that solution because i have to recover the password. So it seems the token access to anilist expires after some time? Is it suppose to? Or is it just an exception?

**Comment by curche** [comment]
>  So it seems the token access to anilist expires after some time? 

Correct. Their tokens last for one year and after that you need to get one more. 
The app doesn't really handle this case separately but I suppose you could make a new "feature request" issue for having that

> Is it suppose to? Or is it just an exception?

Yes and No
Anilist uses OAuth whose spec allows for developers using a service to refresh tokens automagically once the current one expires. Something like how MangaDex (or GitHub too I think?) currently does but with tokens being shortlived (15mins or so, can be refreshed for a month or more and after that new token)
Anilist logins tokens cannot be refreshed and only expire after a year. Which kinda makes sense since its not mango/anime stats for users so its fiiiiine
And so, in a way, AniList is an exception


tl;dr: if you want a better error when anilist login expires after a year, file a new report. (or there's already an issue for that and there's nothing you have to do except wait)

I'll be closing this issue as your issue is resolved
